include/deal.II/base/discrete_time.h

```diff
@@ -78,24 +87,178 @@ DEAL_II_NAMESPACE_OPEN
+ *   are interdependencies between them. For example, if some variable such as
+ *   $x$ depends on the calculation of another variable such as $y$, then $y$
+ *   must be updated before $x$ can be updated. The question arises whether
+ *   time should be incremented before updating state quantities. Multiple
+ *   possibilities exist, depending on program and formulation requirements,
+ *   and possibly the programmer's preferences:
+ *   * Time is incremented before the rest of the updates. In this case, even
+ *     though time is incremented to $t_{n+1}$, not all variables are updated
+ *     yet. During this update phase, $dt$ equals the *previous* time step
+ *     size. *Previous* means that it is referring to the $dt$ of the
+ *     `advance_time()` command that was performed previously. In the
+ *     following example code, we are assuming that `a` and `b` are two state
+ *     variables that need to be updated in this time step.
+ *     @code
+ *       time.advance_time();
+ *       new_a = update_a(a, b, time.get_previous_step_size());
+ *       b = update_b(a, b, time.get_previous_step_size());
+ *       a = new_a;
+ *     @endcode
+ *   * Time is incremented from $t_n$ to $t_{n+1}$ after all variables have
+ *     already been updated for $t_{n+1}$. During the update stage, $dt$ is
+ *     denoted as the *next* time step size. *Next* means that $dt$ of the
+ *     step corresponds to the `advance_time()` command that will happen
+ *     subsequently.
+ *     @code
+ *       new_a = update_a(a, b, time.get_next_step_size());
+ *       b = update_b(a, b, time.get_next_step_size());
+ *       a = new_a;
+ *       time.advance_time();
+ *     @endcode
+ *   * Time is incremented in the middle of the other updates: In this case
+ *     $dt$ would correspond to *next* or *previous* depending of whether it
+ *     is used before or after the call to `advance_time()`.
+ *     @code
+ *       new_a = update_a(a, b, time.get_next_step_size());
+ *       time.advance_time();
+ *       b = update_b(a, b, time.get_previous_step_size());
+ *       a = new_a;
+ *     @endcode
+ *
+ * One thing to note is that, during the update phase, $dt$ is referred to
+ * either **next** or **previous** time step size, depending on whether the
+ * command `advance_time()` has been called yet. The notion of *current* time
+ * step size is ill-defined. In fact, in the update stage the definition of
+ * every variable depends on whether it has been updated yet or not, hence the
+ * name **the inconsistent stage**.
+ *
+ * The following code snippet shows the code sections for the snapshot stage
+ * and the update stage in the context of a complete time-dependent
+ * simulation. This code follows the coding conventions incorporated in the
+ * tutorial examples.
+ * @code
+ * // pre-processing/setup stage {
+ * make_grid();
+ * setup_system();
+ * for (DiscreteTime time(0., 1., 0.1);  // } end pre-processing/setup stage
```

Comment:
```suggestion
 * for (DiscreteTime time(0., 1., 0.1);  // end pre-processing/setup stage
```

Reply:
This seems to be the end brace to `pre-processing/setup stage {`

Reply:
Since it is consistent with the way I denoted the start and end of each stage in `// ...` comments, I am going to keep it as it is now. I hope @bangerth is okay with it.